### PR TITLE
Change Text extraction is not allowed error to warning

### LINKED ISSUE #350: Add check_extractable argument to high_level.extract_text
**Is your feature request related to a problem? Please describe.**
When using `pdfminer.high_level.extract_text` on some files, I get `pdfminer.pdfdocument.PDFTextExtractionNotAllowed: Text extraction is not allowed`

**Describe the solution you'd like**
Add a keyword argument `check_extractable` to `pdfminer.high_level.extract_text`, and pass it to `PDFPage.get_pages`

### PR DESCRIPTION
**Pull request by madhurcodes**
I was using this library for parsing a large set of files and it gave an error halfway through because it encountered such a file. Users should not have to handle this manually in a library meant for parsing PDFs. 

Changes the `Text extraction is not allowed` error thrown to a descriptive warning.

Fixes #350 

**How Has This Been Tested?**

I tested this on an example PDF which threw the mentioned error with the original code and gave the correct text after the change as expected. I cannot share the PDF because I don't own copyright to it.

**Checklist**

- [x] I have added tests that prove my fix is effective or that my feature  works
- [x] I have added docstrings to newly created methods and classes
- [ ] I have optimized the code at least one time after creating the initial version
- [x] I have updated the [README.md](../README.md) or I am verified that this is not necessary
- [x] I have updated the [readthedocs](../docs/source) documentation or I verified that this is not necessary
- [x] I have added a consice human-readable description of the change to [CHANGELOG.md](../CHANGELOG.md)

### CODE REVIEW

File: pdfminer/pdfpage.py [hunk 1]
```diff
@@ -120,15 +121,18 @@ def search(obj, parent):
     @classmethod
     def get_pages(cls, fp,
                   pagenos=None, maxpages=0, password='',
-                  caching=True, check_extractable=True):
```
Comment (pietermarsman) [inline pdfminer/pdfpage.py:123]: Set default to `False` instead of removing.

File: pdfminer/pdfpage.py [hunk 2]
```diff
@@ -120,15 +121,18 @@ def search(obj, parent):
...
-        if check_extractable and not doc.is_extractable:
-            error_msg = 'Text extraction is not allowed: %r' % fp
-            raise PDFTextExtractionNotAllowed(error_msg)
+        # Check if the document allows text extraction.
+        # If not, warn the user and proceed.
+        if not doc.is_extractable:
+            warning_msg = 'This PDF contains a metadata field indicating'\
+                          'that it should not allow text extraction.'    \
+                          'Ignoring this field and proceeding.'
+            warnings.warn(warning_msg, PDFTextExtractionNotAllowedWarning)
```
Comment (pietermarsman) [inline pdfminer/pdfpage.py:135]: Raise error if `check_extractable` and raise warning if not `check_extractable`

File: pdfminer/pdfpage.py (hunk 1 above)
Comment (madhurcodes) [reply pdfminer/pdfpage.py:123]: Done

File: pdfminer/pdfpage.py (hunk 2 above)
Comment (madhurcodes) [reply pdfminer/pdfpage.py]: Done

**Comment by madhurcodes** [comment]
Hey as mentioned in the PR the PDFs I have which trigger this warning are copyright protected so I can't upload them here for testing. I have tested that it raises the correct warning on my machine.

The code is really basic and it should be clear that there is no functional change.

**Comment by pietermarsman** [comment]
I tried changing the content of an sample encrypted document, but apparently its hard to change the password of an encrypted file using only vim :confounded: 

Should have looked at the issue earlier. An example pdf is posted by @Recursing. You could use that one. I also notices that the encrypted samples that are in the repo already are never used for testing, these can also be used. 

A lot of functionality of pdfminer.six is still untested. Having a test suite for all functionality makes it easier to add code, and to change/improve implementations in the future. That's why I'm asking. I do believe that this code works as intended :smile: 

If you have time to add the sample pdf from the issue, please do so. Otherwise I will find some time to do it myself and then merge this. Thanks for all the efforts so far!

**Comment by pietermarsman** [comment]
Thanks @madhurcodes

**Pull request merged**